### DISCUSSION TITLE
Missing preprocessor event handlers

### DISCUSSION THREAD

**Discussion by josephcdb** [badeball/cypress-cucumber-preprocessor#923] [Q&A]
Hi all,
I have encountered an error message and it says it might be a bug, therefore I have to report here.
Anyone can help me resolve this issue?

`Missing preprocessor event handlers (this usally means you've not invoked addCucumberPreprocessorPlugin() or not returned the config object in setupNodeEvents()) (this might be a bug, please report at https://github.com/badeball/cypress-cucumber-preprocessor)

Because this error occurred during a before all hook we are skipping all of the remaining tests.

Although you have test retries enabled, we do not retry tests when before all or after all hooks fail`

<img width="1119" alt="Screenshot 2023-01-12 at 17 27 54" src="https://user-images.githubusercontent.com/21998704/212124357-a032e29d-2c73-42fb-bd33-c6c1bef0e476.png">

**Comment by badeball** [comment]
Did you follow the examples and do what the message suggests?

**Reply by josephcdb** [reply]
Yes I followed the example like https://github.com/badeball/cypress-cucumber-preprocessor and experienced this error.

But I have used setupNodeEvents function under e2e.js and my old cypress plugin was migrated to version 12. I do not have addCucumberPreprocessorPlugin function.

<img width="452" alt="Screenshot 2023-01-12 at 17 42 33" src="https://user-images.githubusercontent.com/21998704/212127915-7b4aa44d-efd5-49f9-abea-0ad8871a7c17.png">

I am not sure where is the issue causing. Your hint would be appreciated.

**Comment by badeball** [comment]
If you're not calling `addCucumberPreprocessorPlugin` anywhere then you're not following the examples. Take a closer look [here](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/quick-start.md#example-setup) and [here](https://github.com/badeball/cypress-cucumber-preprocessor/tree/master/examples).

**Reply by josephcdb** [reply]
Thanks for this link. It really helps me a lot. I will figure it out and have it fixed.

But I have a question. I notice I have two filename -> cypress.config.ts and index.ts under plugins folder. I notice your example do not have index.ts under plugins folder. Can I ignore index.ts and refactor two functions (setupNodeEvents and addCucumberPreprocessorPlugin) on cypress.config.ts instead?

**Reply by badeball** [reply]
Yes, the examples show this.

**Reply by josephcdb** [reply]
Ok, I have deleted index.ts under plugin and move the script into cypress.config.ts.. Now it is working fine. Thank you for the hint. :)
